src/containers/Profile/Profile.js

```diff
@@ -4,32 +4,48 @@ import { connect } from 'react-redux';
 import { NavLink } from 'react-router-dom';
 import EventContainer from '../EventContainer/EventContainer';
 import { setUser } from '../../actions';
+import { updateUser } from '../../thunks/updateUser';
 
-export const Profile = ({ location, user }) => {
-  const [contentEditable, toggleEditable] = useState(false);
-  const buttonText = contentEditable ? 'Save' : 'Edit Profile';
-
+export const Profile = ({ location, user, updateUser }) => {
   if (user.name) {
+    const [contentEditable, toggleEditable] = useState(false);
+    const [name, editName] = useState(user.name);
+    const [bio, editBio] = useState(user.bio);
+    const buttonText = contentEditable ? 'Save' : 'Edit Profile';
+
+    const saveUser = () => {
+      toggleEditable(!contentEditable);
+      contentEditable && updateUser({ name, bio });
+    };
+
     return (
       <div className="Profile">
         <div className="user-profile">
-          <img src={user.image_url} alt="user headshot" className="user-photo" />
+          <img
+            src={user.image_url}
+            alt="user headshot"
+            className="user-photo"
+          />
           <div className="user-bio-section">
             <div className="bio-header">
-              <h1 className="user-name" contentEditable={contentEditable}>
-                {user.name.toUpperCase()}
-              </h1>
+              {!contentEditable && (
+                <h1 className="user-name">{name.toUpperCase()}</h1>
+              )}
+              {contentEditable && (
+                <input className='edit-name-input' value={name.toUpperCase()} onChange={e => editName(e.target.value)} />
+              )}
```

Comment:
You could refactor these && statements into a single ternary here and on lines 45-48